### PR TITLE
Update description of 'max_temp_directory_size'

### PR DESCRIPTION
**Pull request by szarnyasg**
Update description of configuration option

### CODE REVIEW

**Comment by Tishj** [comment]
https://github.com/duckdb/duckdb/pull/10978

The description *should* explain what the default values are, it's a little complicated and this new description doesn't quite cover it. I think docs is a better place to address this

**Review by Tishj — APPROVED** [review]

File: src/include/duckdb/main/settings.hpp [hunk 1]
```diff
@@ -596,7 +596,8 @@ struct StreamingBufferSize {
 struct MaximumTempDirectorySize {
 	static constexpr const char *Name = "max_temp_directory_size";
 	static constexpr const char *Description =
-	    "The maximum amount of data stored inside the 'temp_directory' (when set) (e.g. 1GB)";
+	    "If the `temp_directory` is set to an existing directory, this option defaults to the available disk space on "
```
Comment (Mytherin) [inline src/include/duckdb/main/settings.hpp:599]: This now only describes the default value - should we leave in the description of the setting?

Comment (szarnyasg) [reply src/include/duckdb/main/settings.hpp]: Yes, I pushed an update with this

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**